diff -rupN roundup-1.5.1.orig/roundup/backends/back_postgresql.py roundup-1.5.1/roundup/backends/back_postgresql.py
--- roundup-1.5.1.orig/roundup/backends/back_postgresql.py	2016-01-11 14:53:59.000000000 +0000
+++ roundup-1.5.1/roundup/backends/back_postgresql.py	2016-05-01 10:51:06.786627564 +0000
@@ -93,12 +93,12 @@ def db_command(config, command, database
     conn.set_isolation_level(0)
     cursor = conn.cursor()
     try:
-        for n in range(10):
+        for n in range(100):
             if pg_command(cursor, command):
                 return
     finally:
         conn.close()
-    raise RuntimeError('10 attempts to create database failed')
+    raise RuntimeError('100 attempts to create database failed')
 
 def pg_command(cursor, command):
     '''Execute the postgresql command, which may be blocked by some other
