diff -rupN roundup-1.6.0.orig/roundup/scripts/roundup_server.py roundup-1.6.0/roundup/scripts/roundup_server.py
--- roundup-1.6.0.orig/roundup/scripts/roundup_server.py	2018-07-08 15:35:39.000000000 +0000
+++ roundup-1.6.0/roundup/scripts/roundup_server.py	2018-08-05 18:18:53.625470061 +0000
@@ -692,7 +692,7 @@ class ServerConfig(configuration.Config)
             # This sets the socket to non-blocking. SSL needs a blocking
             # socket, so we do this only for non-SSL connections.
             if hasattr(socket, 'setdefaulttimeout'):
-                socket.setdefaulttimeout(60)
+                socket.setdefaulttimeout(300)
             base_server = BaseHTTPServer.HTTPServer
 
         # obtain request server class
