diff --git a/roundup/scripts/roundup_server.py b/roundup/scripts/roundup_server.py
index b424ec49..58c322b7 100644
--- a/roundup/scripts/roundup_server.py
+++ b/roundup/scripts/roundup_server.py
@@ -724,7 +724,7 @@ class ServerConfig(configuration.Config):
             # This sets the socket to non-blocking. SSL needs a blocking
             # socket, so we do this only for non-SSL connections.
             if hasattr(socket, 'setdefaulttimeout'):
-                socket.setdefaulttimeout(60)
+                socket.setdefaulttimeout(300)
             base_server = http_.server.HTTPServer
 
         # obtain request server class
