diff -rupN roundup-1.6.0.orig/roundup/hyperdb.py roundup-1.6.0/roundup/hyperdb.py
--- roundup-1.6.0.orig/roundup/hyperdb.py	2018-07-04 23:49:25.000000000 +0000
+++ roundup-1.6.0/roundup/hyperdb.py	2018-08-05 18:19:22.708231467 +0000
@@ -257,8 +257,11 @@ class Multilink(_Pointer):
                 try:
                     curvalue.remove(itemid)
                 except ValueError:
-                    raise HyperdbValueError, _('property %s: %r is not ' \
-                        'currently an element')%(propname, item)
+                    # This can occur if the edit adding the element
+                    # produced an error, so the form has it in the
+                    # "no selection" choice but it's not set in the
+                    # database.
+                    pass
             else:
                 newvalue.append(itemid)
                 if itemid not in curvalue:
