diff -rupN roundup-1.5.1.orig/roundup/backends/back_postgresql.py roundup-1.5.1/roundup/backends/back_postgresql.py
--- roundup-1.5.1.orig/roundup/backends/back_postgresql.py	2016-01-11 14:53:59.000000000 +0000
+++ roundup-1.5.1/roundup/backends/back_postgresql.py	2016-05-01 11:05:00.552929864 +0000
@@ -108,23 +108,18 @@ def pg_command(cursor, command):
     '''
     try:
         cursor.execute(command)
-    except psycopg.ProgrammingError, err:
+    except psycopg.DatabaseError, err:
         response = str(err).split('\n')[0]
-        if response.find('FATAL') != -1:
-            raise RuntimeError(response)
-        else:
-            msgs = [
+        if "FATAL" not in response :
+            msgs = (
                 'is being accessed by other users',
-                'could not serialize access due to concurrent update',
-            ]
-            can_retry = 0
-            for msg in msgs:
-                if response.find(msg) == -1:
-                    can_retry = 1
-            if can_retry:
-                time.sleep(1)
-                return 0
-            raise RuntimeError(response)
+                'could not serialize access due to concurrent update'
+            )
+            for msg in msgs :
+                if msg in response :
+                    time.sleep(0.1)
+                    return 0
+        raise RuntimeError (response)
     return 1
 
 def db_exists(config):
