diff -rupN roundup-1.5.1.orig/roundup/hyperdb.py roundup-1.5.1/roundup/hyperdb.py
--- roundup-1.5.1.orig/roundup/hyperdb.py	2016-01-11 14:53:59.000000000 +0000
+++ roundup-1.5.1/roundup/hyperdb.py	2016-05-01 10:52:18.395384323 +0000
@@ -232,8 +232,11 @@ class Multilink(_Pointer):
                 try:
                     curvalue.remove(itemid)
                 except ValueError:
-                    raise HyperdbValueError, _('property %s: %r is not ' \
-                        'currently an element')%(propname, item)
+                    # This can occur if the edit adding the element
+                    # produced an error, so the form has it in the
+                    # "no selection" choice but it's not set in the
+                    # database.
+                    pass
             else:
                 newvalue.append(itemid)
                 if itemid not in curvalue:
