diff --git a/roundup/cgi/client.py b/roundup/cgi/client.py
index a4a6fd4b..b6b6d4f2 100644
--- a/roundup/cgi/client.py
+++ b/roundup/cgi/client.py
@@ -1983,6 +1983,8 @@ class Client:
         """
         try:
             call(*args, **kwargs)
+        except socket.timeout:
+            pass
         except socket.error as err:
             err_errno = getattr (err, 'errno', None)
             if err_errno is None:
