diff -rupN roundup-1.6.0.orig/roundup/cgi/client.py roundup-1.6.0/roundup/cgi/client.py
--- roundup-1.6.0.orig/roundup/cgi/client.py	2018-07-08 15:35:39.000000000 +0000
+++ roundup-1.6.0/roundup/cgi/client.py	2018-08-12 23:19:37.286735714 +0000
@@ -839,14 +839,24 @@ class Client:
             elif self.env.get('HTTP_AUTHORIZATION', ''):
                 # try handling Basic Auth ourselves
                 auth = self.env['HTTP_AUTHORIZATION']
-                scheme, challenge = auth.split(' ', 1)
+                try:
+                    scheme, challenge = auth.split(' ', 1)
+                except ValueError:
+                    # Invalid header.
+                    scheme = ''
+                    challenge = ''
                 if scheme.lower() == 'basic':
                     try:
                         decoded = base64.decodestring(challenge)
                     except TypeError:
                         # invalid challenge
                         pass
-                    username, password = decoded.split(':', 1)
+                    try:
+                        username, password = decoded.split(':', 1)
+                    except ValueError:
+                        # Invalid challenge.
+                        username = ''
+                        password = ''
                     try:
                         # Current user may not be None, otherwise
                         # instatiation of the login action will fail.
