diff -rupN roundup-2.4.0.orig/roundup/cgi/client.py roundup-2.4.0/roundup/cgi/client.py
--- roundup-2.4.0.orig/roundup/cgi/client.py	2024-07-09 13:36:08.000000000 +0000
+++ roundup-2.4.0/roundup/cgi/client.py	2025-01-01 00:43:08.069311363 +0000
@@ -2320,6 +2320,8 @@ class Client:
         """
         try:
             call(*args, **kwargs)
+        except socket.timeout:
+            pass
         except socket.error as err:
             err_errno = getattr(err, 'errno', None)
             if err_errno is None:
