diff -rupN roundup-2.0.0.orig/roundup/cgi/client.py roundup-2.0.0/roundup/cgi/client.py
--- roundup-2.0.0.orig/roundup/cgi/client.py	2020-06-29 14:35:12.000000000 +0000
+++ roundup-2.0.0/roundup/cgi/client.py	2020-07-22 12:07:01.396684267 +0000
@@ -2002,6 +2002,8 @@ class Client:
         """
         try:
             call(*args, **kwargs)
+        except socket.timeout:
+            pass
         except socket.error as err:
             err_errno = getattr (err, 'errno', None)
             if err_errno is None:
